certora/specs/rewardsPreserve.spec

```diff
@@ -0,0 +1,186 @@
+	uint256 claimablePre = getClaimableRewards(e, e.msg.sender, _DummyERC20_rewardToken);
+
+	// Ensure contract has sufficient rewards
+	require _DummyERC20_rewardToken.balanceOf(currentContract) >= claimablePre;
+
+	claimRewardsToSelf(e, _rewards);
+
+	uint256 rewardsBalancePost = _DummyERC20_rewardToken.balanceOf(e.msg.sender);
+	uint256 unclaimedPost = getUnclaimedRewards(e.msg.sender, _DummyERC20_rewardToken);
+	uint256 claimablePost = getClaimableRewards(e, e.msg.sender, _DummyERC20_rewardToken);
+	
+	assert rewardsBalancePost >= rewardsBalancePre, "Rewards balance reduced after claim";
+	uint256 rewardsGiven = rewardsBalancePost - rewardsBalancePre;
+	assert claimablePre == rewardsGiven + unclaimedPost, "Rewards given unequal to claimable";
+	assert claimablePost == unclaimedPost, "Claimable different from unclaimed";
+	assert unclaimedPost == 0;  // Left last as this is an implementation detail
+}
+
+/**
+ * @title Rewards claiming when rewards are insufficient
+/* Ensures rewards are updated correctly after claiming, when there aren't
+ * enough funds.
+ *
+ * - Failed in previous version of the code: job-id=`274946aa85a247149c025df228c71bc1`
+ * - Failure reported in: `https://github.com/bgd-labs/static-a-token-v3/issues/23`
+ * - Passed after fix: job-id=`75c9ee79444b4c8b833a378e5d260599`
+ */
+rule rewardsConsistencyWhenInsufficientRewards() {
+	// Assuming single reward
+	single_RewardToken_setup();
+
+	env e;
+	require e.msg.sender != currentContract;  // Cannot claim to contract
+	require e.msg.sender != _TransferStrategyHarness;
+
+	uint256 rewardsBalancePre = _DummyERC20_rewardToken.balanceOf(e.msg.sender);
+	uint256 claimablePre = getClaimableRewards(e, e.msg.sender, _DummyERC20_rewardToken);
+
+	// Ensure contract does not have sufficient rewards
+	require _DummyERC20_rewardToken.balanceOf(currentContract) < claimablePre;
+
+	claimSingleRewardOnBehalf(e, e.msg.sender, e.msg.sender, _DummyERC20_rewardToken);
+
+	uint256 rewardsBalancePost = _DummyERC20_rewardToken.balanceOf(e.msg.sender);
+	uint256 unclaimedPost = getUnclaimedRewards(e.msg.sender, _DummyERC20_rewardToken);
+	uint256 claimablePost = getClaimableRewards(e, e.msg.sender, _DummyERC20_rewardToken);
+	
+	assert rewardsBalancePost >= rewardsBalancePre, "Rewards balance reduced after claim";
+	uint256 rewardsGiven = rewardsBalancePost - rewardsBalancePre;
+	// Note, when `rewardsGiven` is 0 the unclaimed rewards are not updated
+	assert (
+		( (rewardsGiven > 0) => (claimablePre == rewardsGiven + unclaimedPost) ) &&
+		( (rewardsGiven == 0) => (claimablePre == claimablePost) )
+		), "Claimable rewards changed unexpectedly";
+}
+
+
+/**
+ * @title Only claiming rewards should reduce contract's total rewards balance
+ * Only "claim reward" methods (and `initialize`) should cause the total rewards
+ * balance of `StaticATokenLM` to decline.
+ *
+ * WARNING: `metaDeposit` seems to be vacuous, i.e. **always** fails on a require statement.
```

Comment:
do you know which require?
is this before you used the optimistic hash? was the pessimistic hash the reason for that?